docs/src/configuration/app/web/custom-headers.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Set custom headers on static content
+sidebarTitle: Custom headers
+description: Set custom headers for your static content such as custom content-types or limits to cross-origin usage.
+---
+
+When your app responds to dynamic requests, it can generate headers on the fly.
+If you want to set headers for static content, you can do so in your `web` configuration.
+
+You might want to do so to add custom content-type headers, limit what other sites can embed your content,
+or allow cross origin requests.
+
+Say you want to limit most files to be embedded only on your site,
+but you want an exception for MP3 files.
+And you want to serve both MP3 and MP4 files with the correct content types to avoid [MIME sniffing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#mime_sniffing).
+
+Start by defining a header for files in general:
+
+```yaml {location=".platform.app.yaml"}
+web:
+    locations:
+        "/":
+            ...
+            headers:
+                X-Frame-Options: SAMEORIGIN
+```
+
+This sets the `X-Frame-Options` header to `SAMEORIGIN` for all static files.
+Now your files can only be embedded within your site.
+
+Now set up an exception for MP3 files.
+
+```yaml {location=".platform.app.yaml"}
+web:
+    locations:
+        "/":
+            ...
+            rules:
+                \.mp3$:
+                    headers:
+                      Content-Type: audio/mpeg
+```
+
+This rule sets an explicit content type for files that end in `.mp3`.
+Because specific rules override the general heading configuration,
```

Comment:
I'm not sure I understand what would be clarified by linking to the rules explanation specifically here. This sentence is about this specific configuration. Linking to generic explanation seems like it might be more confusing as it might not be clear what is being explained. 

I can add it above, though I'm not wild about the potential distraction.